SabreTools.DatTools/Rebuilder.cs

```diff
@@ -408,8 +429,32 @@ public class Rebuilder
                     // Get the output archive, if possible
                     Folder outputArchive = GetPreconfiguredFolder(datFile, date, outputFormat);
 
-                    // Now rebuild to the output file
-                    outputArchive.Write(fileStream, outDir, (item as Rom).ConvertToBaseFile());
+                    
+                    BaseFile itemRom = (item as Rom).ConvertToBaseFile();
+
+                    // change filename inside folder or zip to orignal name
+                    //
+                    // e.g. taking TOSEC files as input and replace them with 
+                    // No-Intro Names from DAT (which are more correct and shorter),
+                    // but leave names inside as original TOSEC Name to find them quicker
+                    if (baseReplace)
+                        itemRom.Filename = datItem.GetName();
+
+                    if (archivesAsFiles) {
```

Comment:
Improper formatting of curly braces. Please follow the style guide and surrounding code.